homeassistant/components/telegram_bot/__init__.py

```diff
@@ -993,7 +1038,12 @@ class BaseTelegramBotEntity:
 
     def __init__(self, hass, config):
         """Initialize the bot base class."""
-        self.allowed_chat_ids = config[CONF_ALLOWED_CHAT_IDS]
+        self.allowed_chat_ids = {
+            x if isinstance(x, int) else x[CONF_CHAT_ID]: None
+            if isinstance(x, int)
+            else x.get(CONF_USER_ID)
+            for x in config[CONF_ALLOWED_CHAT_IDS]
```

Comment:
This is quite unreadable. Transform the CONF_ALLOWED_CHAT_IDS to a dictionary during schema validation instead.